### PR TITLE
yum: add integration test for 'update foo*'

### LINKED ISSUE #45250: Improve yum update and wildcard
##### SUMMARY
Currently the only implementation of wildcard '*' in yum module is `yum update -y`
See the line [here 
](https://github.com/ansible/ansible/blob/6be42a2a0ec6afb1e8a2f5f48bef4e9c1588dd34/lib/ansible/modules/packaging/os/yum.py#L1140)

##### ISSUE TYPE
- Change this to accept also a package name, etc
```yaml
yum:
  name: 'python*'
  state: latest
```
and this would actualy run a `yum update -y python*` not a `yum update -y`

##### COMPONENT NAME

- [lib/ansible/modules/packaging/os/yum.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/packaging/os/yum.py)

##### ANSIBLE VERSION

```paste below
ansible 2.6.3
  config file = None
  configured module search path = [u'/home/user/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.15 (default, May 16 2018, 17:50:09) [GCC 8.1.1 20180502 (Red Hat 8.1.1-1)]

```
##### CONFIGURATION

I am using default configuration

##### OS / ENVIRONMENT

Running Ansible on Fedora 28, targeting Vagrant centos/7 box

##### ADDITIONAL INFORMATION

I definitely think that there's such a use case where you need to update/upgrade some part of packages for one "family" like in my case Python. Also the implementation already has `exclude: kernel*`
Current workaround is to run `shell` module, running it via yum would give us a much cleaner code and better usage.

```yaml
  tasks:
    - name: Update Python packages
      shell: yum update -y python*
      args:  
        warn: false
      register: yum_update
      changed_when: '"No packages marked for update" not in yum_update.stdout'
```

### PR DESCRIPTION
**Pull request by mkrizek**
##### SUMMARY

Fixes https://github.com/ansible/ansible/issues/45250

Shows that #45250 works as expected.

##### ISSUE TYPE

- Test Pull Request

##### COMPONENT NAME

yum

##### ANSIBLE VERSION

```paste below
devel
```

##### ADDITIONAL INFORMATION

### CODE REVIEW

File: test/integration/targets/yum/tasks/repo.yml [hunk 1]
```diff
@@ -588,3 +588,58 @@
...
+            - "'msg' in yum_result"
+            - "'rc' in yum_result"
+            - "'results' in yum_result"
+  always:
+    - name: Clean up
+      yum:
+        name: foo,foo-bar,bar
+        state: absent
+
+  when: ansible_pkg_mgr == 'yum'
```
Comment (maxamillion) [inline test/integration/targets/yum/tasks/repo.yml:645]: Any specific reason to not allow this to run for the `dnf` backend also?

Comment (mkrizek) [reply test/integration/targets/yum/tasks/repo.yml:645]: No, just copy&paste without realizing that. Do you want me to create a follow up PR for that?

Comment (maxamillion) [reply test/integration/targets/yum/tasks/repo.yml:645]: @mkrizek yes please :)

Comment (mkrizek) [reply test/integration/targets/yum/tasks/repo.yml:645]: https://github.com/ansible/ansible/pull/48607

**Pull request merged**